### DISCUSSION TITLE
Why is cloudflare module workers preset not recommended?

### DISCUSSION THREAD

**Discussion by awdr74100** [nitrojs/nitro#2214] [Q&A]
In the official documentation it is mentioned that the cloudflare module workers preset is not recommended, but if you want to use nitro to build a pure backend api, shouldn't you use this preset? Is my understanding incorrect in any way?

I'm going to build a pure backend api rather than a full-stack application with pages.

Any help from anyone is appreciated.

**Comment by pi0** [comment]
Hi. This is mainly because cloudflare is pushing to move all new users to pages format. Also using pages you have free CI/CD and free traffic for serving static assets.

/cc @dario-piotrowicz

**Reply by awdr74100** [reply]
Thank you for your prompt replies every time! It really helped me a lot!

**Reply by dario-piotrowicz** [reply]
Thanks for the ping @pi0 🙂 

Yeah anyways I'm just reiterating/confirming what Pooya just said (with just some added context details which hopefully might be helpful 🙂)

The Module workers preset, since it uses workers, needs to make use of [Workers Sites](https://developers.cloudflare.com/workers/configuration/sites/) in order to serve static assets. Workers Sites is practically unofficially deprecated in favour of Pages so that's the reason why the module workers preset is not recommended (not because we specifically want people not to use workers, but because we want people not to use Workers Sites (which is the only way to serve static assets with workers)).

Besides the DX improvements Pooya mentioned, the static assets serving is different between Workers Sites and Pages, in Pages static assets are served for free without invoking the underlying worker, with Workers Sites however, the worker itself needs to programmatically do the assets fetching, so that results in compute time, which after the free tier incurs actual costs for the developer.

**Reply by mayfieldiv** [reply]
Hi all, it now seems like Cloudflare has chosen "Workers Static Assets" over Cloudflare Pages as their current plan for a unified future.

Quote from both a [recent blog post](https://blog.cloudflare.com/builder-day-2024-announcements/#static-asset-hosting) and [new docs](https://developers.cloudflare.com/workers/static-assets): "We plan to bridge the gaps between Workers and Pages and provide ways to migrate your Pages projects to Workers."

Also the [docs](https://developers.cloudflare.com/workers/configuration/sites/) for the (deprecated) Workers Sites say "You should use [Workers Static Assets](https://developers.cloudflare.com/workers/static-assets) to host full-stack applications instead of Workers Sites. Do not use Workers Sites for new projects."

**Reply by thebrownfox** [reply]
Is there any disadvantage using pages (and their functions) instead of workers? Seems like it's the same thing except the CI/CD thing.

**Reply by pi0** [reply]
features like durable objects (usable for sqlite and websocket with pub/sub) are only available in workers format for now. You can check [compatibility matrix](https://developers.cloudflare.com/workers/static-assets/compatibility-matrix/) to compare.
